geoportailv3/static/js/mymaps/mymapsdirective.js

```diff
@@ -38,6 +39,98 @@ app.module.directive('appMymaps', app.mymapsDirective);
  * @constructor
  * @export
  */
-app.MymapsDirectiveController = function() {};
+app.MymapsDirectiveController = function() {
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.defaultTitle_ = 'Map without title';
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.modifying = false;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.choosing = false;
+
+  /**
+   * @type {Object?}
+   * @export
+   */
+  this.mymap = null;
+};
+
+
+/**
+ * Closes the current map.
+ * @export
+ */
+app.MymapsDirectiveController.prototype.close = function() {
+  // TODO ensure that modifications are saved.
+  this.mymap = null;
+};
+
+
+/**
+ * Creates a new map.
+ * @export
+ */
+app.MymapsDirectiveController.prototype.create = function() {
+  this.onChosen();
+};
+
+
+/**
+ * Open a map. Actually opens the map selector.
+ * @export
+ */
+app.MymapsDirectiveController.prototype.choose = function() {
+  this['choosing'] = true;
+};
+
+
+/**
+ * Called when a map is choosen.
+ * @export
+ */
+app.MymapsDirectiveController.prototype.onChosen = function() {
+  this.mymap = {
+    id: 'mymap-987654321',
+    title: this.defaultTitle_
+  };
+  this.choosing = false;
+};
+
+
+/**
+ * Start the modification:
+ *  - opens the modification modal,
+ *  - set the values to form inputs.
+ * @export
+ */
+app.MymapsDirectiveController.prototype.modify = function() {
+  this['newTitle'] = this['mymap']['title'];
+  this['newDescription'] = this['mymap']['description'];
+  this['modifying'] = true;
```

Comment:
You should declare all the properties in the constructor and use the dot notation.